public/pages/AnomalyCharts/components/FeatureChart/FeatureChart.tsx

```diff
@@ -125,72 +154,86 @@ export const FeatureChart = (props: FeatureChartProps) => {
-                  fill: darkModeEnabled() ? 'red' : '#D5DBDB',
-                }}
-              />
-            ) : null}
-            <Axis
-              id="left"
-              title={props.featureDataSeriesName}
-              position="left"
-              showGridLines
-            />
-            <Axis id="bottom" position="bottom" tickFormat={timeFormatter} />
-            <LineSeries
-              id="featureData"
-              name={props.featureDataSeriesName}
-              xScaleType={ScaleType.Time}
-              yScaleType={ScaleType.Linear}
-              xAccessor={CHART_FIELDS.PLOT_TIME}
-              yAccessors={[CHART_FIELDS.DATA]}
-              data={featureData}
-            />
-          </Chart>
-          {showCustomExpression ? (
-            <CodeModal
-              title={props.feature.featureName}
-              subtitle="Custom expression"
-              code={JSON.stringify(props.feature.aggregationQuery, null, 4)}
-              getModalVisibilityChange={() => true}
-              closeModal={() => setShowCustomExpression(false)}
+      <div
+        style={{
+          height: '200px',
+          width: '100%',
+          opacity: showLoader ? 0.2 : 1,
+        }}
+      >
+        <Chart>
+          <Settings
+            showLegend
+            showLegendExtra={false}
+            //TODO: research more why only set this old property will work.
+            showLegendDisplayValue={false}
+            legendPosition={Position.Right}
+            theme={FEATURE_CHART_THEME}
+          />
+          {props.feature.featureEnabled ? (
+            <RectAnnotation
+              dataValues={props.annotations || []}
+              id="annotations"
+              style={{
+                stroke: darkModeEnabled() ? 'red' : '#D5DBDB',
+                strokeWidth: 1,
+                opacity: 0.8,
+                fill: darkModeEnabled() ? 'red' : '#D5DBDB',
+              }}
             />
           ) : null}
-        </div>
-      ) : (
-        <EuiEmptyPrompt
-          style={{ maxWidth: '45em' }}
-          body={
-            <EuiText>
-              <p>{`There is no data to display for feature ${props.feature.featureName}`}</p>
-            </EuiText>
-          }
-        />
-      )}
+          {props.feature.featureEnabled &&
+          props.showFeatureMissingDataPointAnnotation &&
+          props.detectorEnabledTime
+            ? [
+                <LineAnnotation
+                  id="featureMissingAnnotations"
+                  domainType={AnnotationDomainTypes.XDomain}
+                  dataValues={getFeatureMissingDataAnnotations(
+                    props.showFeatureMissingDataPointAnnotation
+                      ? props.rawFeatureData
+                      : props.featureData,
+                    props.detectorInterval.interval,
+                    getFeatureMissingAnnotationDateRange(
+                      props.dateRange,
+                      props.detectorEnabledTime
```

Comment:
So we will only show missing data alerts from latest detector enabled time? Is it possible we show alerts for data before?

Reply:
Yes, only since latest enabled time. Ideally, we should show alert after detector 1st enabled time, and we should not show data during time period when detector disabled. However, there is no way to keep the 1st enabled time, also no way to keep track of which time period detector is disabled. If detector is disabled/enabled multiple times, it is hard for us to ignore the disabled time periods. That's why we only show alerts since latest enabled time.